nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/core/ElasticSearchAdminImpl.java

```diff
@@ -93,12 +99,27 @@
 
     /**
      * Init the admin service, remote configuration if not null will take precedence over local embedded configuration.
+     * @deprecated since 9.1
      */
     public ElasticSearchAdminImpl(ElasticSearchLocalConfig localConfig, ElasticSearchRemoteConfig remoteConfig,
             Map<String, ElasticSearchIndexConfig> indexConfig) {
         this.remoteConfig = remoteConfig;
         this.localConfig = localConfig;
         this.indexConfig = indexConfig;
+        this.clientInitService = null;
+        connect();
+        initializeIndexes();
```

Comment:
Don't repeat this code, make it call the new constructor passing `null` as an argument for `clientInitService`.